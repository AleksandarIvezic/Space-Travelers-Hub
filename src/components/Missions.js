import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import fetchMissions from '../helpers/fetchMissions';
import { getMissions } from '../redux/missions/missions';

const Missions = () => {
  const dispatch = useDispatch()

  useEffect(async () => {
    dispatch(getMissions(await fetchMissions()))
  }, [])

  const data = useSelector((state) => state.missions)

  return (
    <div>Missions
    </div>
  );
};

export default Missions;
