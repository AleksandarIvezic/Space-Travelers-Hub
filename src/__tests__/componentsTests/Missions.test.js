import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/storeConfig';
import Missions from '../../components/Missions';

describe('Missions', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}><Missions /></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
