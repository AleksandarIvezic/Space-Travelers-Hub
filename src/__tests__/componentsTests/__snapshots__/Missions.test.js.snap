// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions renders correctly 1`] = `
<ul
  className="m-0 p-0"
>
  <div
    className="container"
  >
    <table
      className="table table-striped table-bordered"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Mission
          </th>
          <th
            scope="col"
          >
            Description
          </th>
          <th
            scope="col"
          >
            Status
          </th>
          <th
            scope="col"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</ul>
`;
